### DISCUSSION TITLE
How to automatically open the last file after restarting zed ?

### DISCUSSION THREAD

**Discussion by arfbt** [zed-industries/zed#11838] [Feature Requests]
How to automatically open the last file after restarting zed ? When I reopened zed, it went blank. That's too bad.

**Comment by Twig6943** [comment]
Bump. How can I do this?

**Comment by Twig6943** [comment]
@arfbt Do  `"restore_on_startup": "last_session"`

**Reply by TheSilverNimbus** [reply]
This doesn't work for me. I tried opening four different files and then closed the window. When I open Zed again, it starts with a blank file (not the four files I had opened earlier).

**Reply by arfbt** [reply]
same, doesn't work

**Comment by VKondakoff** [comment]
Not an answer. This works only on project level. And don’t works if there is no project open ad you have an opened file, which does not  belong to any project. After Zed restart this file will be lost.

**Reply by Kamek437** [reply]
Yeah I'd like it to open and be exactly like I left it, even if opening a new file.

**Reply by Twig6943** [reply]
> Yeah I'd like it to open and be exactly like I left it, even if opening a new file.

same here

**Comment by ReZeroS** [comment]
When closing windows containing untitled/unsaved files, Zed displays a save confirmation dialog. If users accidentally click "Don't Save", these temporary documents are permanently lost.

**Comment by TheSilverNimbus** [comment]
I don't know if this is the thread for it (or if I should open a new Issue), but when I have a few files open in a specific configuration (see image), I'd like that session to be restored to that same configuration.

_(**Note:** The screenshot attached to this is merely a representation of what I hope to be restored when I restart Zed, in this example. I'm aware that the files shown in the image are not saved files that were opened in the editor.)_

![Screenshot_20250528_151101](https://github.com/user-attachments/assets/0de334c2-d023-4bdb-a45a-b1dce730ea27)

**Comment by aplatypus** [comment]
I think it is relevant to comment that Zed seems to open the files open with the last project closed-off.  At least that is how it appears to me the first time I open a Zed session.  I like to keep a active Zed (edit) session for working notes and a scratch pad I call `junk.md`.  This Zed instance is shared across all my desktops and activities.

To resume my on-going editing session, I only need to select the `junk.md` project.  That has been my 80% solution to this challenge.  The default Zed behaviour seems to be an empty window or a "recent" project.  

My understanding of  the use case for this question: 

- [`How to automatically open the last file after restarting zed ?`](https://github.com/zed-industries/zed/discussions/11838#top)

... is that folk would like to see the last project editing session resumed with the files open again.  In one respect that appears to be what Zed wants to do too.  What could satisfy this use case if for either ...

- Use a command launcer to resume your edit session of choice like my junk example:
  `zed $(HOME)/tmp/junk.md`
- Zed (or a plugin) to set a default project, or project list,  to resume when first started
- Zed uses a config setting to identify the default project(s) to open

Personally, I think I like a launcer notion where there's a `sh` script to launch one or more named projects.

One footnote is that projects only seem to be identified by directory or (consitiuent) file name.  In my world project directories can and do move around.  I find it desirable to have a name paired with a location.  That is a different topic/use case, although it seems to have a bearing on the topic from my perspective (or tastes).

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
